Python/sysmodule.c

```diff
@@ -116,6 +116,11 @@ sys_breakpointhook(PyObject *self, PyObject *const *args, Py_ssize_t nargs, PyOb
         /* The breakpoint is explicitly no-op'd. */
         Py_RETURN_NONE;
     }
+    envar = _PyMem_RawStrdup(envar);
```

Comment:
Gosh, I'm really becoming uncomfortable with all the undocumented, internal, underscored functions.  It makes it much harder to understand CPython's implementation since now I have to go read the source for `_PyMem_RawStrdup`.  This isn't the only case of it, and I know we can't fix that in this PR, but I do think we need to do better to document the private C API.

I would suggest adding a comment above this explaining the POSIX semantics, so that a future reader will understand why a copy is being made.